apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,218 @@
+// Copyright 2020-2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitHubIdentityProviderPhase string
+
+const (
+	// GitHubPhasePending is the default phase for newly-created GitHubIdentityProvider resources.
+	GitHubPhasePending GitHubIdentityProviderPhase = "Pending"
+
+	// GitHubPhaseReady is the phase for an GitHubIdentityProvider resource in a healthy state.
+	GitHubPhaseReady GitHubIdentityProviderPhase = "Ready"
+
+	// GitHubPhaseError is the phase for an GitHubIdentityProvider in an unhealthy state.
+	GitHubPhaseError GitHubIdentityProviderPhase = "Error"
+)
+
+type GitHubAllowAllUsersToLoginSpec string
+
+const (
+	// GitHubAllowAllUsersToLogin means any GitHub user is allowed to log in to this Pinniped,
+	// regardless of their organization membership or lack thereof.
+	GitHubAllowAllUsersToLogin GitHubAllowAllUsersToLoginSpec = "Allow"
+
+	// GitHubDoNotAllowAllUsersToLogin means only those users with membership in specified GitHub
+	// organizations are allowed to log in.
+	GitHubDoNotAllowAllUsersToLogin GitHubAllowAllUsersToLoginSpec = "Deny"
+)
+
+// GitHubIdentityProviderStatus is the status of an GitHub identity provider.
+type GitHubIdentityProviderStatus struct {
+	// Phase summarizes the overall status of the GitHubIdentityProvider.
+	//
+	// +kubebuilder:default=Pending
+	// +kubebuilder:validation:Enum=Pending;Ready;Error
+	Phase GitHubIdentityProviderPhase `json:"phase,omitempty"`
+
+	// Conditions represents the observations of an identity provider's current state.
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// GitHubApiConfig allows configuration for GitHub Enterprise Server
+type GitHubApiConfig struct {
+	// Host is required only for GitHub Enterprise Server.
+	// Defaults to using GitHub's public API (github.com).
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// TLS configuration for GitHub Enterprise Server.
+	// +optional
+	TLS *TLSSpec `json:"tls,omitempty"`
+}
+
+// GitHubUsernameClaimSpec allows the user to specify which attribute(s) from GitHub to use for the downstream username.
+// See the response for https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user.
+type GitHubUsernameClaimSpec string
```

Comment:
Is "Spec" a good suffix for this string?  There are a lot of Spec structs in Kubernetes related software, they are usually a struct.

Reply:
I forget why I chose that suffix, but you're right it doesn't really need to be here. 

Done, will push fix